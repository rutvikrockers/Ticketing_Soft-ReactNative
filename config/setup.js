// create our store for testing redux components
import configureMockStore from 'redux-mock-store';
const thunk = require('redux-thunk');

// Setup adapter to work with enzyme 3.2.0
const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');

export const mockStore = configureMockStore([thunk.default]);

Enzyme.configure({ adapter: new Adapter() });