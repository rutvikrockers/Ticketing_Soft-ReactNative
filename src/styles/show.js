import {
    StyleSheet,
  } from 'react-native';
  import Profile from '../components/user/show';
  const styles = StyleSheet.create({
    ...Profile,
    container: {
        flex: 1,
        backgroundColor: '#fff',
      },
      headArea: {    
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginBottom: 20,
        marginTop: 20, 
        padding: 20 
      },
      headText: {
        fontSize: 30,
        fontWeight: 'bold',
        color: '#2d2d2d',
      },
      linkarea: {
          paddingHorizontal: 20,
          marginBottom: 10
      },
      hr:{
        borderTopWidth: 1,
        borderTopColor: '#eeeeee',
        marginLeft: 20,    
        marginBottom:30    
      },
      linkItem: {
        marginBottom:20
      },
      linkText: {
        color: '#2c2c2c',
        fontSize: 18
      }
  });

  module.exports = styles;