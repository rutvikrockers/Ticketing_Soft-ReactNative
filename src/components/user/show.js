import React, {Component} from 'react';
import {Text, ScrollView, TouchableHighlight, View, Image,StyleSheet} from 'react-native';
import {connect} from 'react-redux';

import Footer from '../../components/shared/footer';
import Home from '../../views/home';
import {Link, Redirect} from 'react-router-native';
import Followers from '../../components/user/followers'
import Filters from '../../components/user/filters';

import Settings from '../../components/user/Settings';
import styles from '../../styles/show';


class Profile extends Component {
  
  render() {
    var props = this.props;
    var match = props.match;
    return (
      
      <View style={styles.container}>
      <View style={styles.headArea}>
        <Text style={styles.headText}>Account</Text>
      </View>
      <View style={styles.linkarea}>
        
          <Link to="/userprofile" style={styles.linkItem}><Text style={styles.linkText}>UserProfile</Text></Link>
          <TouchableHighlight style={styles.linkItem}>
              <Text style={styles.linkText}>Investment History</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.linkItem}>
              <Text style={styles.linkText}>Payment</Text>
          </TouchableHighlight>
      </View>
      <View style={styles.hr}></View>
      <View style={styles.linkarea}>
          <TouchableHighlight style={styles.linkItem}>
              <Text style={styles.linkText}>Invite Friends</Text>
          </TouchableHighlight>
          <TouchableHighlight style={styles.linkItem}>
              <Text style={styles.linkText}>Linked Accounts</Text>
          </TouchableHighlight>
      </View>
      <View style={styles.hr}></View>
      <View style={styles.linkarea}>
     
      
      <Link to="/settings" style={styles.linkItem}><Text style={styles.linkText}>Settings</Text></Link>
          
          
      </View>
     
        <Footer />
    </View> 
    );
  }
}


const mapStateToProps = (state) => {
  return {
    user: state.user
  }
}

export default connect(mapStateToProps)(Profile);

