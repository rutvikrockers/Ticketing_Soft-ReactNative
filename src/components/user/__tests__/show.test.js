import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { mockStore } from '../../../../config/setup';
import toJson from 'enzyme-to-json';
import { default as Show } from '../show';

const render = (props) => shallow(
    <Provider store={mockStore({})}>
        <Show {...props} />
    </Provider>
);

describe('Show', () => {
    let component;
    beforeEach(() => {
        component = render({});
    });

    it('matches its snapshot', () => {
        expect(toJson(component, { mode: 'deep' })).toMatchSnapshot();
    });

    it('renders', () => {
        expect(component.find(Show).length).toBe(1);
    });
});