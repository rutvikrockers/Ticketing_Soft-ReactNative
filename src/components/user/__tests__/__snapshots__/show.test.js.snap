// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show matches its snapshot 1`] = `<Connect(Profile) />`;
